### PR TITLE
Fixes #38055 - Fix future-dated subscriptions

### PR DESCRIPTION
**Pull request by jeremylenz**
#### What are the changes introduced in this pull request?

Problem 1:
Candlepin's `GET owners/{owner_key}/products` endpoint, called by `Katello::Resources::Candlepin::Product.all`, recently added a new param, `active`. When the param is not included, it apparently causes Candlepin to omit (some?) products with future-dated subscriptions. Sending the param `active=include` should restore the previous behavior. This affects Candlepin version 4.4.15 and above.

Problem 2:
The "Add Subscriptions" page does not display future-dated subscriptions, e.g. subscriptions whose _start date_ is in the future. Sending the `add_future` param should fix that.

#### Considerations taken when implementing this change?

I don't think this should break anything, since the param would just be ignored on previous CP versions.

#### What are the testing steps for this pull request?

1. Obtain a manifest with a future-dated subscription and no other subscriptions

2. Import into a new organization

Without this PR: Manifest import fails with
```
Validation failed: Subscription can't be blank
```
From another org with a valid manifest imported, the "Add Subscriptions" page does not show any future-dated subscriptions available to add.

After: 
Manifest import succeeds
You can see any future-dated subscriptions on the Add Subscriptions page. Compare with access.redhat.com > Subscription allocations > your manifest > Add subscriptions. The available subscriptions in Satellite should include future-dated ones just like this page does.

### CODE REVIEW

File: test/scenarios/scenario_test.rb [hunk 1]
```diff
@@ -21,17 +21,17 @@ def test_manifest_import
...
       manifest_path = File.join(::Katello::Engine.root, 'test', 'fixtures', 'files', 'manifest_small_modified.zip')
       @support.import_manifest(@org.label, manifest_path)
       sleep 2
       @support.import_products(@org, manifest_path)
       @org.reload
 
-      assert_equal 3, @org.products.length
-      assert_equal 9, @org.product_contents.length
+      assert_equal 342, @org.products.length
+      assert_equal 9597, @org.product_contents.length
```
Comment (jeremylenz) [inline test/scenarios/scenario_test.rb:34]: This is what made tests pass locally for me. But I am not sure if this test is correct or not. It does make sense that the number of imported products would vary now that I'm sending include=active, but I wouldn't expect numbers this big. I looked inside the fixture manifest files and didn't see 342 of anything..

File: app/lib/katello/resources/candlepin/product.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ module Candlepin
       class Product < CandlepinResource
         class << self
           def all(owner_label, included = [])
-            products = JSON.parse(Candlepin::CandlepinResource.get(path(owner_label) + "?#{included_list(included)}", self.default_headers).body)
+            products = JSON.parse(Candlepin::CandlepinResource.get(path(owner_label) + "?active=include&#{included_list(included)}", self.default_headers).body)
```
Comment (parthaa) [inline app/lib/katello/resources/candlepin/product.rb:7]:
```suggestion
           url = path(owner_label) + "?active=include" + "&#{included_list(included)}"
            products = JSON.parse(Candlepin::CandlepinResource.get(url, self.default_headers).body)
```

slightly more readable imp

File: test/scenarios/scenario_test.rb (hunk 1 above)
Comment (jeremylenz) [reply test/scenarios/scenario_test.rb:34]: @qcjames53 is getting completely different numbers. Maybe it's a difference between what's in our test databases.

**Review by qcjames53 — APPROVED** [review]
I was able to confirm that during manifest import the `active=include` string works on Candlepin version 4.4.19 and allows inclusion of future-dated subscriptions. As mentioned in another comment, the manifest import test seems to produce inconsistent results from Candlepin depending on the run; this behavior was observed in the existing test so it is not a regression. I asked the team for input on this and we deemed it okay because testing Candlepin API responses is outside of the scope of this test.

Jeremy, you're clear to merge after pushing the smaller recordings. ACKed

@sjha4 thank you for taking a look at this :)

**Pull request merged**